Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
-            raise DemistoException('Authorization Error: make sure API Key is correctly set')
-        else:
-            raise e
-    return "ok"
+            raise DemistoException('Authorization Error: Make sure client credentials are correctly set') from e
+        raise
 
 
-def get_raw_events(client: Client, time_of_last_event: str) -> list:
-    """
-       helper function that is used in get-events and fetch-events to get the actual events and sort them
-       Args:
-           client (Client): DG client
-           time_of_last_event: time of last ingested event
-       Returns:
-           list: list of events
-    """
-    outcome = []
-    event_list = []
-    if not time_of_last_event:
-        time_of_last_event_datetime = datetime.now() - timedelta(hours=1)
-        time_of_last_event_str = datetime_to_string(time_of_last_event_datetime)
-    else:
-        temp_time: datetime = arg_to_datetime(arg=time_of_last_event, required=True)  # type: ignore[assignment]
-        time_of_last_event_str = temp_time.isoformat(sep=' ', timespec='milliseconds')
-    current_time = datetime_to_string(datetime.now())
-    events = client.get_events(time_of_last_event_str, current_time)
-    events = {} if events is None else events
-    for field_names in events.get("fields"):
-        outcome.append(field_names.get('name'))
-    for event in events.get("data"):
-        result = dict(zip(outcome, event))
-        event_list.append(result)
-    event_list.sort(key=lambda item: (item.get("inc_mtime"), item.get("dg_guid")))
-    return event_list
-
-
-def get_events_command(client: Client, args: dict) -> Tuple[list, CommandResults]:
+def fetch_events(client: Client, limit: int | None = None) -> tuple[EventsGenerator, dict]:
     """
-        Implement the get_events command
-        Args:
-            client (Client): DG client
-            args (dict): Command arguments
-        Returns:
-            list: list of events
-            commandresults: readable output
+    Args:
+        client (Client): Digital Guardian client.
+        limit (int | None): Optional value to limit the number of yielded events.
+    Returns:
+        tuple[EventsGenerator, dict]: Events and last run dictionary.
     """
-    event_list = get_raw_events(client, "")
-    limit = int(args.get("limit", 1000))
-    if limit:
-        event_list = event_list[:limit]
-    if not event_list:
-        hr = "No events found."
-    else:
-        hr = tableToMarkdown(name='Test Event', t=event_list)
-    demisto.debug(f'get events command that ran with the limit: {limit}')
-    return event_list, CommandResults(readable_output=hr)
+    demisto.debug('Fetching events')
+    raw_response = client.export_events()
+
+    demisto.info(f"Pulled {raw_response['total_hits']} events from {VENDOR}")
+    events = create_events_for_push(raw_response, limit)
+
+    last_run = {key: value for key, value in raw_response.items() if key in ('bookmark_values', 'search_after_values')}
```

Comment:
in a first reading I find this line surprising, what's special about these two values? a comment here would make it easier for future maintainers

Reply:
The exact meaning of those two fields is not clear to me nor to the TPM at the moment. 

I will further investigate and get in touch with someone from Digital Guardian if needed.

Reply:
I will get in touch with the relevant individuals and add an informative comment

Reply:
I added a (somewhat vague) comment, I will add more specifics in case I get any updates.